cmd/up.go

```diff
@@ -142,20 +152,45 @@ func isGKE(disco discovery.DiscoveryInterface) (bool, error) {
 	return false, nil
 }
 
-func generateMongoDBSecret(cli kubernetes.Interface, objs []*unstructured.Unstructured) error {
-	var mongoNS string
-	for _, obj := range objs {
-		if obj.GetKind() == "Deployment" && obj.GetName() == "mongodb" {
-			mongoNS = obj.GetNamespace()
-			break
-		}
-	}
-	if mongoNS == "" {
-		return fmt.Errorf("Unable to find the namespace of the MongoDB database")
+func mongoSecretExists(c kubecfg.ApplyCmd) (*unstructured.Unstructured, bool, error) {
+	gvk := schema.GroupVersionKind{Version: "v1", Kind: "Secret"}
+	rc, err := clientForGroupVersionKind(c.ClientPool, c.Discovery, gvk, Kubeapps_NS)
+	if err != nil {
+		return nil, false, err
 	}
-	err := populateSecretWithPasswords(cli, mongoNS, "mongodb", []string{"mongodb-password", "mongodb-root-password"})
+	pwFields := []string{"mongodb-password", "mongodb-root-password"}
+	prevSec, err := rc.Get(SecretID)
 	if err != nil {
-		return err
+		if k8sErrors.IsNotFound(err) {
+			return nil, false, nil
+		} else {
+			return nil, true, err
+		}
+	} else {
+		if prevSec.Object["data"] != nil {
+			prevPw := prevSec.Object["data"].(map[string]interface{})
+			for _, p := range pwFields {
+				if prevPw[p] == nil {
+					return nil, true, fmt.Errorf("secret %s already exists but it doesn't contain the expected key %s", SecretID, p)
+				}
+			}
+		} else {
+			return nil, true, fmt.Errorf("secret %s already exists but it doesn't contain any expected key", SecretID)
+		}
+		return prevSec, true, nil
+	}
+}
+
+func buildMongoDBSecret(pw map[string]string) *unstructured.Unstructured {
```

Comment:
again, if the `name` and `namespaces` are params the function becomes generic (the same happens with `deleteMongoDBSecret`)